ggml.c

```diff
@@ -5290,8 +5290,8 @@ struct ggml_tensor * ggml_mul_mat(
         struct ggml_context * ctx,
         struct ggml_tensor  * a,
         struct ggml_tensor  * b) {
-    GGML_ASSERT(ggml_can_mul_mat(a, b));
-    GGML_ASSERT(!ggml_is_transposed(a));
+    // GGML_ASSERT(ggml_can_mul_mat(a, b));
+    // GGML_ASSERT(!ggml_is_transposed(a));
```

Comment:
These asserts should not be removed. If you hit them, then there is most likely something wrong with the input data

Reply:
I think I found the issue, 
![Screenshot 2024-06-03 at 12 02 51 PM](https://github.com/ggerganov/llama.cpp/assets/70980118/557dbb2b-ef76-48cc-9373-6cdebb2774eb)
tensor b should be `[1024, 576] -> [4096, 576]`

**For LLAVA these dimensions are right:**
![Screenshot 2024-06-03 at 12 04 29 PM](https://github.com/ggerganov/llama.cpp/assets/70980118/37ddb708-0a7c-4f0e-9458-c8e8c4fab09a)

**But for phi3v we need the mm_projector weight tensor to be `[4096, 576]`:**
![Screenshot 2024-06-03 at 12 05 11 PM](https://github.com/ggerganov/llama.cpp/assets/70980118/44b6b0a9-f8e8-4589-a02c-aade45244d2a)

Any idea on how to fix this?

Reply:
I see there is a reference to these dims here @ggerganov 
https://github.com/ggerganov/llama.cpp/blob/bde7cd3cd949c1a85d3a199498ac98e78039d46f/examples/llava/clip.cpp#L698

Reply:
This might be it. I think it will have to be implemented in `clip.cpp`
https://huggingface.co/microsoft/Phi-3-vision-128k-instruct/blob/main/image_embedding_phi3_v.py#L219:~:text=%23%20(num_crops%2C%2012%2C%202,num_img_tokens%2C%201024*4)